core/src/main/java/org/commonjava/indy/core/expire/ScheduleManager.java

```diff
@@ -474,14 +476,16 @@ public synchronized void setRepoTimeouts( final StoreKey key, final String path
             return;
         }
 
-        if ( deploy.getRepoTimeoutSeconds() > 0 )
-        {
-            final int timeout = deploy.getRepoTimeoutSeconds();
+        Integer timeout = deploy.getRepoTimeoutSeconds();
 
-            cancel( new StoreKeyMatcher( key, STORE_JOB_TYPE ), path );
+        cancel( new StoreKeyMatcher( key, STORE_JOB_TYPE ), STORE_JOB_TYPE );
 
-            scheduleStoreExpiration( key, timeout );
+        if ( timeout == null || timeout <= 0 )
```

Comment:
I think `if ( timeout != null && timeout > 0 ) { scheduleStoreExpiration( key, timeout ); }` is easier logic to follow. It's also less likely to result in side effects.